pkg/artifact/registry.go

```diff
@@ -55,7 +64,92 @@ func (a registry) LocationDataForLaunchable(stanza manifest.LaunchableStanza) (*
 		return location, verificationData, nil
 	}
 
-	return nil, auth.VerificationData{}, util.Errorf("Version key in launchable stanza not implemented")
+	if a.registryURL == nil {
+		return nil, auth.VerificationData{}, util.Errorf("No artifact registry configured and location field not present on launchable %s", launchableID)
+	}
+
+	return a.fetchRegistryData(launchableID, stanza.Version)
+}
+
+type RegistryResponse struct {
+	ArtifactURL          string `json:"artifact_url"`
+	ManifestURL          string `json:"manifest_url"`
+	ManifestSignatureURL string `json:"manifest_signature_url"`
+	BuildSignatureURL    string `json:"build_signature_url"`
+}
+
+func (a registry) fetchRegistryData(launchableID manifest.LaunchableID, version manifest.LaunchableVersion) (*url.URL, auth.VerificationData, error) {
```

Comment:
note: this function isn't yet complete, pending knowing the query structure we want for the registry fetch.

1) launchableID not used yet
2) no OS version tag yet